lib/puma/cluster.rb

```diff
@@ -608,6 +664,50 @@ def wait_workers
       end
     end
 
+    def check_process_terminated(pid)
+      Process.wait(pid, Process::WNOHANG)
+    rescue Errno::ECHILD
+      begin
+        Process.kill(0, pid)
+      rescue Errno::ESRCH, Errno::EPERM
+        return true
+      else
+        false
+      end
+    end
+
+    def promote_mold
+      missing_workers = @options[:workers] - @workers.size
+      return if missing_workers <= 0
+
+      # if the active mold got reaped, remove it here
+      @mold = nil unless @tracked_molds.include?(@mold)
+
+      # if the mold is not pinging, send it a TERM and let it die next iteration
+      if @mold && @mold.ping_timeout <= Time.now
+        log "- Mold timed out, terminating"
```

Comment:
Let's include some additional info here (specifically the PID), similar to: https://github.com/puma/puma/blob/0bf46dbfbe585a8c034009ce14cbdde7772cea97/lib/puma/cluster.rb#L126